rules/windows/defense_evasion_whitespace_padding_in_command_line.toml

```diff
@@ -0,0 +1,41 @@
+[metadata]
+creation_date = "2021/07/30"
+maturity = "production"
+updated_date = "2021/07/30"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies process execution events where the command line value contains superfluous whitespace characters. Attackers
+may attempt to evade signature-based detections by padding their malicious command with unnecessary whitespace
+characters. These observations should be investigated for malicious behavior.
+"""
+from = "now-9m"
+index = ["winlogbeat-*", "logs-endpoint.events.*", "logs-windows.*"]
+language = "eql"
+license = "Elastic License v2"
+name = "Whitespace Padding in Process Command Line"
+note = """## Triage and analysis
+
+- Analyze the command line of the process in question for evidence of malicious code execution.
+- Review the ancestry and child processes spawned by the process in question for indicators of further malicious code execution."""
+references = ["https://twitter.com/JohnLaTwC/status/1419251082736201737"]
+risk_score = 47
+rule_id = "e0dacebe-4311-4d50-9387-b17e89c2e7fd"
+severity = "medium"
+tags = ["Elastic", "Host", "Windows", "Threat Detection", "Defense Evasion"]
+timestamp_override = "event.ingested"
+type = "eql"
+
+query = '''
+process where event.type in ("start", "process_started") and
+  process.command_line regex ".*[ ]{20,}.*"
```

Comment:
It might be worthwhile to expand this to include multiple multi-space gaps as well

```suggestion
  process.command_line regex ".*[ ]{20,}.*" or 
  process.command_line regex ".*([ ]{5,}){3,}.*"
```

![image](https://user-images.githubusercontent.com/16747370/127692282-9fd0c57f-0eb6-4566-845f-be189d52e504.png)

Reply:
`.*([ ]{5,}){3,}.*` is not equivalent to `.*[ ]{15,}.*`

The former looks for 3 or more groups of 5+ space chunks where as the latter is looking for 15+ whitespace contiguous chunk. 

This is a commonly known attempt to evade detection that is just as applicable as the original approach IMO.

Reply:
> `.*([ ]{5,}){3,}.*` is not equivalent to `.*[ ]{15,}.*`

It is though. Unless you meant this: `.*(.*[ ]{5,}){3,}.*`

Reply:
ah, yea, one step better actually: `.*(.*[ ]{5,}.*){3,}.*`

Reply:
I think that's actually redundant

Reply:
to definitively break into separate space chunks `.*(.*[ ]{5,}[^ ]*){3,}.*`

Reply:
Can someone please update the suggestion with the final recommendation for the regex expression along with a description of what it matches on? I'll review that before merging the new rule

Reply:
I updated it to reflect my last suggestions, if that LGTY @rw-access and @threat-punter

Reply:
Can you add a comment to explain what the expression matches on? I think that will be incredibly useful to anyone reviewing the rule and understanding what it matches on.

Reply:
FWIW I still like the original version from @threat-punter the most.
The new one is fine, but does seem like a different rule and might not even catch the original behavior that motivated the rule.

Reply:
Just to point out, it is not replacing the original query, just expanding on it, since it is also common evasion behavior. It doesn't seem to make sense to make a separate rule for it  because it is just as fitting with the rule as the original - _Whitespace Padding in Process Command Line_

If there is this much concern for it though, it is not worth stopping the rule, so feel free to skip it if that is what it takes to get it merged

Reply:
ah good point thanks. I'm okay either way then and I'll punt to @threat-punter to decide

Reply:
@threat-punter 

>Can you add a comment to explain what the expression matches on? I think that will be incredibly useful to anyone reviewing the rule and understanding what it matches on.

I added explanatory comment to the query suggestion above and a [suggestion](https://github.com/elastic/detection-rules/pull/1392/files#r686348176) to expand the description to include this behavior as well.

Reply:
👍